lib/stages/visRawReader.cpp

```diff
@@ -351,6 +351,7 @@ void visRawReader::main_thread() {
         // Set the dataset ID to the updated value
         dset_id_t& ds_id = ((visMetadata*)(out_buf->metadata[frame_id]->metadata))->dataset_id;
         ds_id = get_dataset_state(ds_id);
+        ((visMetadata*)(out_buf->metadata[frame_id]->metadata))->dataset_id = ds_id;
```

Comment:
I don't get it. Isn't this achieved by using a reference to this two line above?

Reply:
Hmm, I guess that is right. I had to go through a few iterations before I got things to work, and I thought this was one problem along the way, not noticing it is a reference. I'll remove this commit and check it still works.